test/e2e/runner/cli_provisioner.go

```diff
@@ -205,6 +205,8 @@ func (cli *CLIProvisioner) provision() error {
 	}
 	cli.Engine = eng
 
+	cli.EnsureArcResourceGroup()
```

Comment:
O.K., lets add this after L30 above (after we've called `CreateGroupWithRetry` and received a nil err):

```
now := fmt.Sprintf("now=%v", time.Now().Unix())
cli.Account.ResourceGroup = r := ResourceGroup{
	Name:     cli.Config.Name,
	Location: cli.Config.Location,
	Tags: map[string]string{
		"now": now,
	},
}
```

Then we can get rid of the `now` var in the `CreateGroup` method, and get rid of the whole "assign ResourceGroup property to account instance" code.

I think that's the easiest path forward. You can then get rid of the new duplicated "create resource group func" and basically do exactly what you were doing before, without the surprise Account.ResourceGroup override. :)